swing/src/net/sf/openrocket/gui/scalefigure/RocketPanel.java

```diff
@@ -573,6 +573,8 @@ private void updateExtras() {
 		FlightConditions conditions = new FlightConditions(curConfig);
 		warnings.clear();
 
+		extraText.setCurrentConfig(curConfig);
```

Comment:
This is the line that actually fixes the bug.

The rest is refactoring to make intent more clear.